core/evm.go

```diff
@@ -65,7 +65,7 @@ func (hooks) OverrideNewEVMArgs(args *vm.NewEVMArgs) *vm.NewEVMArgs {
 	if rules.IsShanghai {
 		args.BlockContext.Random = new(common.Hash)
 		args.BlockContext.Random.SetBytes(args.BlockContext.Difficulty.Bytes())
-		args.BlockContext.Difficulty = nil
+		args.BlockContext.Difficulty = new(big.Int)
```

Comment:
aren't we expecting this to be 1 [here](https://github.com/ava-labs/coreth/blob/libevm/plugin/evm/block_verification.go#L100)?

Reply:
in the header yes, but not in the block context, which is the handoff point to the EVM interpreter.
The interpreter will start using the "Random" opcode, which will read the value of "Random"
The value of Difficulty will actually be unused (Which is how this was previously working), but I still figure it's better to set it the same way as upstream.